src/api.ts

```diff
@@ -11,6 +11,8 @@ interface Env {
 // env object is injected by JSVM
 declare let env: Env;
 
+// TODO: rewrite api.js to named parameters?
```

Comment:
IMO; current interface is okay, as we can indicate the order of arguments match https://nomicon.io/Proposals/system-methods.

Most APIs, except for promise APIs, have a short param list. For promise APIs, these many params are painful. I think for these APIs, we can consider rust sdk's high level promise API, see whether we can build it in js sdk (Maybe in 2.0)

Reply:
Agree. And this API should never change, let's keep it as it is for now.